src/main/java/net/dries007/tfc/objects/blocks/wood/BlockSupport.java

```diff
@@ -292,10 +292,8 @@ public boolean canPlaceBlockOnSide(@Nonnull World world, @Nonnull BlockPos pos,
         }
         else
         {
-            if (!isConnectable(world, pos, side.getOpposite()))
-            {
-                return false;
-            }
+            if (!isConnectable(world, pos, side.getOpposite())) return false;
+            if (!isThreeTall(world, pos.offset(side.getOpposite()))) return false;
             int distance = getHorizontalDistance(side, world, pos);
```

Comment:
This is only checking on one side. Better to implement that logic on `getHorizontalDistance` and only return a value if both sides are at least three tall.